src/logical/builder_proxy.rs

```diff
@@ -13,16 +13,17 @@ use crate::{
 
 /// Trait capturing builder proxies that use plain string (used for parsing in logical layer)
 pub trait LogicalColumnBuilderProxy<'a, 'b>: ColumnBuilderProxy<String> {
-    /// create a new [`LogicalColumnBuilderProxy`] which nests its physical target equivalent [`BuilderProxy`][crate::physical::builder_proxy::PhysicalBuilderProxyEnum].
+    /// Create a new [`LogicalColumnBuilderProxy`] which nests its physical target equivalent [`BuilderProxy`][crate::physical::builder_proxy::PhysicalBuilderProxyEnum].
     ///
+    /// This offers a representation of the parsing from a given logical type to the physical type, without exposing details from one layer to the other.
```

Comment:
First of all, this is not something that belongs on the constructor, but (if at all) on the trait itself. Also, it does not “offer a representation of the parsing”, it _does_ the parsing while properly encapsulating typing details in each of the layers.